baseapp/auth/saml/serviceprovider.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package saml
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/crewjam/saml"
+	"github.com/pkg/errors"
+)
+
+type samlError struct {
+	Err          error
+	ResponseCode int
```

Comment:
This is the recommended response code, not the response code from the IdP or something?